application/src/main/java/org/togetherjava/tjbot/commands/moderation/WhoIsCommand.java

```diff
@@ -0,0 +1,263 @@
+            if (showServerSpecificInfo) {
+                return handleWhoIsMember(event, member, profile);
+            } else {
+                return handleWhoIsUser(event, user, profile);
+            }
+        }).queue();
+    }
+
+    @CheckReturnValue
+    private static @NotNull ReplyAction handleWhoIsUser(final @NotNull SlashCommandEvent event,
+            final @NotNull User user, final @NotNull User.Profile profile) {
+
+        StringBuilder descriptionBuilder =
+                new StringBuilder().append(handleUserIdentification(user))
+                    .append("\n**Is bot:** ")
+                    .append(user.isBot())
+                    .append(handleUserFlags(user.getFlags()))
+                    .append("\n**Registration date:** ")
+                    .append(DATE_TIME_FORMAT.format(user.getTimeCreated()));
+
+        EmbedBuilder embedBuilder =
+                generateEmbedBuilder(event, user, profile, profile.getAccentColor()).setAuthor(
+                        user.getName(), user.getEffectiveAvatarUrl(), user.getEffectiveAvatarUrl())
+                    .setDescription(descriptionBuilder);
+
+        return event.replyEmbeds(embedBuilder.build())
+            .addActionRow(Button.of(ButtonStyle.LINK, "discord://-/users/" + user.getId(),
+                    "Click to see profile"));
+    }
+
+    @CheckReturnValue
+    private static @NotNull ReplyAction handleWhoIsMember(final @NotNull SlashCommandEvent event,
+            final @NotNull Member member, final @NotNull User.Profile profile) {
+        User user = member.getUser();
+
+        Color memberColor = member.getColor();
+        Color effectiveColor = (null == memberColor) ? profile.getAccentColor() : memberColor;
+
+        StringBuilder descriptionBuilder =
+                new StringBuilder().append(handleUserIdentification(user))
+                    .append(handleVoiceState(member))
+                    .append("\n**Is bot:** ")
+                    .append(user.isBot())
+                    .append(handlePossibleBooster(member))
+                    .append(handleUserFlags(user.getFlags()))
+                    .append("\n**Join date:** ")
+                    .append(DATE_TIME_FORMAT.format(member.getTimeJoined()))
+                    .append("\n**Registration date:** ")
+                    .append(DATE_TIME_FORMAT.format(user.getTimeCreated()))
+                    .append("\n**Roles:** ")
+                    .append(formatRoles(member));
+
+
+        EmbedBuilder embedBuilder = generateEmbedBuilder(event, user, profile, effectiveColor)
+            .setAuthor(member.getEffectiveName(), member.getEffectiveAvatarUrl(),
+                    member.getEffectiveAvatarUrl())
+            .setDescription(descriptionBuilder);
+
+
+        return event.replyEmbeds(embedBuilder.build())
+            .addActionRow(Button.of(ButtonStyle.LINK, "discord://-/users/" + user.getId(),
+                    "Click to see profile"));
+    }
```

Comment:
Hard to see during review but is this method a 1:1 copy of the other method? Only with different strings? The embed and button part look almost identical.

Maybe we can extract some stuff into helper methods or get rid of the duplication by other means. Not a big deal though, im okay with this for now.

Reply:
Struggled a ton with this.

Issue is simple, to create a user friendly order I put certain flags in a certain order. (join date before registration date a.e)
So if we compare:

```diff
               new StringBuilder().append(handleUserIdentification(user))
+                    .append(handleVoiceState(member))
                    .append("\n**Is bot:** ")
                    .append(user.isBot())
+                    .append(handlePossibleBooster(member))
                    .append(handleUserFlags(user.getFlags()))
+                    .append("\n**Join date:** ")
+                    .append(DATE_TIME_FORMAT.format(member.getTimeJoined()))
                    .append("\n**Registration date:** ")
                    .append(DATE_TIME_FORMAT.format(user.getTimeCreated()))
+                    .append("\n**Roles:** ")
+                    .append(formatRoles(member));
```

You notice that the differences are everywhere, and I can't keep readable order (unless doing weird af trickery with Map's/Lists) while making it maintainable.

So this means, when adding a field it has to be done twice unfortunately.

Reply:
I guess you could get rid of some of the duplication though, no? Not talking about the embed itself now, you convinced me regarding this. But what about the code shortly before and after the embed building? I.e. maybe just move the embed-building into two helper methods and keep the rest in a single method?

But yeah, if thats also not easily possible, its fine.

Reply:
Why not add a `sendEmbedWithUserProfile` or similar, where u move this duplicated code into.

Reply:
Since you spoiled me with the correct name, sure will create it.
Originally I wasn't sure how to name it, but that method sounds good to me, so sure will create it.

Reply:
Yes it was